### DISCUSSION TITLE
S0 Counter with ESP32

### DISCUSSION THREAD

**Discussion by miwi17null6** [arendst/Tasmota#22001] [Q&A]
hello, I have configured an esp32 so that it recognizes the S0 pulses (1000IMP/KWH - 30ms per pulse). However, too many pulses are counted. Why is that? The difference is not that big. Around 100 pulses per day. But per month it adds up to over 3000 pulses.

Script:
>D
>B
->sensor53 r
>M 1
+1,4,c,1,-10,S0
1,1-0:1.8.0*255(@1000,Energy,KWh,Energy,3)
1,1-0:1.7.0*255(@0.01667,Power,W,Power,0)
#

Tasmota firmware is up to date

ESP32 NODE MCU 38PIN model

Can anyone help?

**Comment by sfromis** [comment]
That sounds like something happening on the hardware level, outside of Tasmota. Potentially, tweaking the debounce time of the meter definition might make a difference, but such would depend on what happens at the hardware level.
